pkg/contract/chain.go

```diff
@@ -100,30 +182,39 @@ func GetBlockchainEndpoints(
 		wsEndpoint  string
 	)
 	switch {
+	case chainSpec.BlockchainName != "":
+		sc, err := app.LoadSidecar(chainSpec.BlockchainName)
+		if err != nil {
+			return "", "", fmt.Errorf("failed to load sidecar: %w", err)
+		}
+		networkName := network.Name()
+		if sc.Networks[networkName].BlockchainID == ids.Empty {
+			// look into the cluster deploys
+			for k := range sc.Networks {
+				sidecarNetwork, err := app.GetNetworkFromSidecarNetworkName(k)
+				if err == nil {
+					if sidecarNetwork.Kind == network.Kind && sidecarNetwork.Endpoint == network.Endpoint {
```

Comment:
nit: we should make a func that we can compare if 2 networks are equal and make it part of the the network.go